### PR TITLE
get average by Yevhen Sahach

### PR DESCRIPTION
**Pull request by eugeneSahach**

### CODE REVIEW

File: src/calculateAverage.js [hunk 1]
```diff
@@ -37,7 +37,35 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    if (options === null) {
+      return undefined;
+    } else if (options === undefined) {
```
Comment (YuriiHoliuk) [inline src/calculateAverage.js]: unneeded else, the same below

File: src/calculateAverage.js [hunk 2]
```diff
@@ -37,7 +37,35 @@ function applyCalculateAverage() {
...
+      return this.reduce((acc, elem) => acc + elem, 0) / this.length;
+    }
+
+    if (options.hasOwnProperty('propertyName')
+      && this.some(elem => !!elem[options.propertyName]
+        && typeof elem[options.propertyName] === 'number')) {
```
Comment (YuriiHoliuk) [inline src/calculateAverage.js]: Do you really need it if you have if inside reduce? Will be code without this check faster or slower?

File: src/calculateAverage.js [hunk 3]
```diff
@@ -37,7 +37,35 @@ function applyCalculateAverage() {
...
+          && typeof elem[options.propertyName] === 'number') {
+          counter++;
+          return acc + elem[options.propertyName];
+        }
+
+        return acc;
+      }, 0) / counter;
+    }
+
+    if (options.hasOwnProperty('accumulator')
```
Comment (YuriiHoliuk) [inline src/calculateAverage.js]: hasOwnProperty is not necessary in this case

**Review by YuriiHoliuk — CHANGES_REQUESTED** [review]

File: src/calculateAverage.js (hunk 2 above)
Comment (yevhen-sahach) [reply src/calculateAverage.js]: yes, I need it in order to avoid NaN as result of my reduce

**Review by irenhh — APPROVED** [review]

**Pull request closed without merging**